acceptance/openstack/networking/v2/extensions/layer3/l3_scheduling_test.go

```diff
@@ -0,0 +1,82 @@
+//go:build acceptance || networking || layer3 || router
+// +build acceptance networking layer3 router
+
+package layer3
+
+import (
+	"testing"
+
+	"github.com/gophercloud/gophercloud/acceptance/clients"
+	networking "github.com/gophercloud/gophercloud/acceptance/openstack/networking/v2"
+	"github.com/gophercloud/gophercloud/acceptance/tools"
+	"github.com/gophercloud/gophercloud/openstack/common/extensions"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/agents"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/layer3/routers"
+	th "github.com/gophercloud/gophercloud/testhelper"
+)
+
+func TestLayer3RouterScheduling(t *testing.T) {
+	client, err := clients.NewNetworkV2Client()
+	th.AssertNoErr(t, err)
+
+	_, err = extensions.Get(client, "l3_agent_scheduler").Extract()
+	if err != nil {
+		t.Skip("Extension l3_agent_scheduler not present")
+	}
+
+	network, err := networking.CreateNetwork(t, client)
+	th.AssertNoErr(t, err)
+	defer networking.DeleteNetwork(t, client, network.ID)
+
+	subnet, err := networking.CreateSubnet(t, client, network.ID)
+	th.AssertNoErr(t, err)
+	defer networking.DeleteSubnet(t, client, subnet.ID)
+
+	router, err := CreateRouter(t, client, network.ID)
+	th.AssertNoErr(t, err)
+	defer DeleteRouter(t, client, router.ID)
+	tools.PrintResource(t, router)
+
+	routerInterface, err := CreateRouterInterfaceOnSubnet(t, client, subnet.ID, router.ID)
+	tools.PrintResource(t, routerInterface)
+	th.AssertNoErr(t, err)
+	defer DeleteRouterInterface(t, client, routerInterface.PortID, router.ID)
+
+	// List hosting agent
+	allPages, err := routers.ListL3Agents(client, router.ID).AllPages()
+	th.AssertNoErr(t, err)
+	hostingAgents, err := routers.ExtractL3Agents(allPages)
+	th.AssertNoErr(t, err)
+	th.AssertEquals(t, len(hostingAgents) > 0, true)
```

Comment:
There's a [AssertIntGreaterOrEqual()](https://github.com/gophercloud/gophercloud/blob/017c152/testhelper/convenience.go#L366) helper function you could use for comparing values.

Reply:
done